test/Tests/src/Semantic/Equals_Spec.enso

```diff
@@ -203,5 +205,15 @@ spec =
                 dupl_tree = tree.deep_copy
                 Test.with_clue "Seed sed to 42" (tree == dupl_tree).should_be_true
 
```

Comment:
I was thinking that we should have tests for `Map` equality to avoid regression on this. But I realised we do have one in `Map_Spec.enso`, so that seems good.

A bit tempting if it would be worth to include some test generating a Java `HashMap` to check if equality of _polyglot maps_ works too. If you think it's worth it maybe it would be nice to add it.

Reply:
I left `Map_Spec.enso` unchanged in this PR.